commcare/src/main/resources/webapp/js/directives.js

```diff
@@ -431,7 +431,9 @@
 
                 if (scope.$parent.selectedConfig !== undefined) {
                     scope.downloadingCases = true;
-                    scope.$apply();
+                    if (!scope.$$phase) {
+                        scope.$apply();
+                    }
```

Comment:
According to Angular wiki, wrapping $apply into such condition is considered an anti-pattern. 
https://github.com/angular/angular.js/wiki/When-to-use-$scope.$apply%28%29

Perhaps we should investigate more throughly why $apply is called here and why we need it at all and do small refactor if necessary.